indra/llui/llbutton.cpp

```diff
@@ -396,6 +421,30 @@ bool LLButton::postBuild()
     return LLUICtrl::postBuild();
 }
 
+void LLButton::onVisibilityChange(bool new_visibility)
+{
+    mFontBuffer.reset();
+    return LLUICtrl::onVisibilityChange(new_visibility);
+}
+
+void LLButton::reshape(S32 width, S32 height, bool called_from_parent)
+{
+    S32 delta_width = width - getRect().getWidth();
+    S32 delta_height = height - getRect().getHeight();
+
+    if (delta_width || delta_height || sForceReshape)
+    {
+        LLUICtrl::reshape(width, height, called_from_parent);
```

Comment:
Won't it call `LLButton::dirtyRect()` indirectly, so font buffer will be reset twice?

Reply:
You are correct, will remove this.